dispatch.go

```diff
@@ -1,13 +1,24 @@
 package rpc
 
-import "sync"
+import (
+	"errors"
+	"sync"
+)
 
 type DecodeNext func(interface{}) error
-type ServeHook func(DecodeNext) (interface{}, error)
-type ServeNotifyHook func(DecodeNext) error
+
+type ServeHookDescription struct {
+	Args func() interface{}
+	Func func(arg interface{}) (ret interface{}, err error)
```

Comment:
I know this was my original name from slack last night, but should we call it something more descriptive?  `Handler`? `Hook`?  I am terrible at naming, so feel free to suggest what's best.